core/src/main/java/io/questdb/cairo/AbstractIntervalDataFrameCursor.java

```diff
@@ -63,6 +63,16 @@ protected static long search(ReadOnlyColumn column, long value, long low, long h
                 low = mid + 1;
             else if (midVal > value)
                 high = mid;
+            else if (midVal > 0) {
```

Comment:
we probably don't need to check that value is positive, timestamp can also be 0 and negative. Need to search for same values regardless

Reply:
Ah sorry, I mean `mid > 0` to prevent ever doing `column.getLong(-1)`

Reply:
loop is walking backwards and can potentially step over the file boundary, so the test need to be in the loop itself

Reply:
Fixed.